cdap-app-fabric/src/main/java/io/cdap/cdap/app/metrics/WorkflowMetrics.java

```diff
@@ -46,4 +49,16 @@ public void gauge(String metricName, long value) {
     programMetricsContext.gauge(metricName, value);
     workflowMetricsContext.gauge(metricName, value);
   }
+
+  @Override
+  public Metrics child(Map<String, String> tags) {
+    return new WorkflowMetrics(programMetricsContext.childContext(tags), workflowMetricsContext.childContext(tags));
+  }
+
+  @Override
+  public Map<String, String> getTags() {
+    Map<String, String> result = new HashMap<>();
+    result.putAll(programMetricsContext.getTags());
```

Comment:
does this mean the context returns a modifiable map? Seems like we should change it to be unmodifiable.

Reply:
Removed the `WorkflowMetrics` class.